### ISSUE TITLE
Type checker reports code inside `sys.platform` check as unreachable

### ISSUE THREAD

**Issue by MikhailArkhipov** [microsoft/pyright#410]
**Describe the bug**
Type checker reports valid code as unreachable

**To Reproduce**
```python
import sys
if sys.platform == 'win32':
    x = 1
```
`x = 1` is reported as unreachable while actually it is. `sys.version` works OK.

**Expected behavior**
No error report on code reachable on Windows.

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/12820357/70590811-aff82980-1b88-11ea-9291-07eb6a323479.png)

**Comment by erictraut** [comment]
This is the "as designed" behavior currently. By default, pyright assumes that code is platform-agnostic, so platform-specific code is treated as unreachable. You can override this behavior and specify a specific platform in the pyrightconfig.json setting if you want.

There are files that conditionally define variables or perform imports based on sys.platform, so pyright needs to make an assumption about which platform the code will eventually run on.

If you think we should change the behavior, let's discuss further.
